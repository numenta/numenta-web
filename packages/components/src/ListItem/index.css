/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._marker {
  composes: ml4 from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: pl1 from 'tachyons';
}


.markerCircle {
  composes: _marker;
  list-style-type: circle;
}
.markerDecimal {
  composes: _marker;
  list-style-type: decimal;
}
.markerDisc {
  composes: _marker;
  list-style-type: disc;
}
.markerIcon {
  composes: list from 'tachyons';
}
.markerSquare {
  composes: _marker;
  list-style-type: square;
}

.copy {
  composes: lh-copy from 'tachyons';
  composes: measure from 'tachyons';
  composes: measure-wide-l from 'tachyons';
}

.li {
  /* */
}
