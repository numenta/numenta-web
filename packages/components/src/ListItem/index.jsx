// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'

const markers = {
  circle: styles.markerCircle,
  decimal: styles.markerDecimal,
  disc: styles.markerDisc,
  icon: styles.markerIcon,
  square: styles.markerSquare,
}


/**
 *
 */
const ListItem = ({children, copy, marker}) => {
  const classes = [styles.li]

  if (copy) classes.push(styles.copy)
  if (marker) classes.push(markers[marker])

  return (
    <li className={classNames(...classes)}>
      {children}
    </li>
  )
}

ListItem.propTypes = {
  children: React.PropTypes.node.isRequired,
  copy: React.PropTypes.bool,
  marker: React.PropTypes.oneOf(Object.keys(markers)),
}

ListItem.defaultProps = {
  copy: true,
}

export default ListItem
