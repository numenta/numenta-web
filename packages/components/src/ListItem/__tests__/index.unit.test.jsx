import React from 'react'
import renderer from 'react-test-renderer'

import ListItem from '../../ListItem'


describe('ListItem React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <ListItem>
        List Item
      </ListItem>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('Copy', () => {

    it('Renders correctly', () => {
      const component = renderer.create(
        <ListItem copy={true}>
          List Item
        </ListItem>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('Renders Non correctly', () => {
      const component = renderer.create(
        <ListItem copy={false}>
          List Item
        </ListItem>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Marker', () => {

    it('Renders Circle correctly', () => {
      const component = renderer.create(
        <ListItem marker="circle">
          List Item
        </ListItem>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('Renders Decimal correctly', () => {
      const component = renderer.create(
        <ListItem marker="decimal">
          List Item
        </ListItem>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('Renders Disc correctly', () => {
      const component = renderer.create(
        <ListItem marker="disc">
          List Item
        </ListItem>
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

  })

})
