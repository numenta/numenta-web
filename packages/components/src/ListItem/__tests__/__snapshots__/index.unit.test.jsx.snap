exports[`ListItem React component Copy Renders Non correctly 1`] = `
<li
  className="li">
  List Item
</li>
`;

exports[`ListItem React component Copy Renders correctly 1`] = `
<li
  className="li copy">
  List Item
</li>
`;

exports[`ListItem React component Marker Renders Circle correctly 1`] = `
<li
  className="li copy markerCircle">
  List Item
</li>
`;

exports[`ListItem React component Marker Renders Decimal correctly 1`] = `
<li
  className="li copy markerDecimal">
  List Item
</li>
`;

exports[`ListItem React component Marker Renders Disc correctly 1`] = `
<li
  className="li copy markerDisc">
  List Item
</li>
`;

exports[`ListItem React component Renders Default correctly 1`] = `
<li
  className="li copy">
  List Item
</li>
`;
